### ISSUE TITLE
Support for the save operation (creating tarball of a Docker image)

### ISSUE THREAD

**Issue by gmolto** [indigo-dc/udocker#74]
Hi,

udocker supports loading container images (created via the _docker save_ command). However it does not support saving container images into a tarball that can be later imported by udocker itself.

This functionality would be used when you do not have constant access neither to a Docker installation nor to Docker Hub. You would use udocker to pull once from Docker Hub and later export the image that could be loaded by other udocker deployments.

To sum up, is _udocker save_ planning to be implemented? We would really much appreciate that functionality.

**Comment by jorge-lip** [comment]
Hi,
Yes it is in the plans and will try to push it to the next release. 
Meanwhile you may also try to create a tarball of the container file system tree and import it with udocker import.
To make the tarball just use tar and the location of the container can be obtained with 
udocker inspect -p container-id

**Comment by gmolto** [comment]
Hi Jorge,

Thank you for the reply. Just to be clear: We want to export the _image_ file system not the _container_ file system, that is, we would like udocker to support the functionality of _docker save_ (saving an image to a tar archive). We are not interested in the functionality of _docker export_ (exporting a container's filesystem as a tar archive).

Regards.

**Comment by zoobab** [comment]
"We are not interested in the functionality of docker export (exporting a container's filesystem as a tar archive)."

But I am. Will you accept a patch if I make the feature?

**Comment by HarisShahid1** [comment]
Hi,
We (@CVPCORP) are really interested in udocker having the save functionality as well. We are using udocker to run from within an AWS Lambda function which, can pull an image from docker hub and then save it to a tar just like docker allows. The image tar is then uploaded to our S3 bucket. We do not need udocker to load an existing saved tar image so we don't really need to have the load functionality as well. Just need save.

Our use case is basically for the AWS Lambda function to serve as an intermediary which has access to download images from docker hub and then upload it to S3 from where it is then pulled and AV scanned before it is uploaded to an internal docker registry for consumption within our organization.

We need this functionality ASAP and would really appreciate it if someone can make it happen (yesterday). Please do let us know if we can help expedite the process in any way shape or form.

**Comment by jorge-lip** [comment]
Hi,
Save is implemented in [1.1.4](https://github.com/indigo-dc/udocker/tree/v1.1.4).  
Closing this issue, if needed please reopen.

**Comment by zoobab** [comment]
Thanks will make some testing...
